core/src/java/main/org/jaxen/expr/DefaultFunctionCallExpr.java

```diff
@@ -170,25 +168,20 @@ public Object evaluate(Context context) throws JaxenException
         Function func = context.getFunction(namespaceURI,
                 prefix,
                 getFunctionName());
-        List paramValues = evaluateParams(context);
+        List<Object> paramValues = evaluateParams(context);
 
         return func.call(context, paramValues);
     }
 
-    public List evaluateParams(Context context) throws JaxenException
+    public List<Object> evaluateParams(Context context) throws JaxenException
     {
-        List paramExprs = getParameters();
+        List<Expr> paramExprs = getParameters();
         int paramSize = paramExprs.size();
 
-        List paramValues = new ArrayList(paramSize);
-
-        for (int i = 0; i < paramSize; ++i)
-        {
-            Expr eachParam = (Expr) paramExprs.get(i);
-
-            Object eachValue = eachParam.evaluate(context);
+        List<Object> paramValues = new ArrayList<Object>(paramSize);
 
-            paramValues.add(eachValue);
+        for (Expr eachParam : paramExprs) {
```

Comment:
foreach might be clearer. There's no proof it's faster.

Reply:
You are right. This is possibly a flawed assumption on my part, but still it's also not slower. The compiler should optimize both to the same speed.